gym/f110_gym/envs/f110_env.py

```diff
@@ -311,6 +322,48 @@ def reset(self, poses):
         obs, reward, done, info = self.step(action)
         return obs, reward, done, info
 
+    def add_obstacles(self, n_obstacles=4, obstacle_size=[0.5, 0.5]):
+        """
+        Adds a set number of obstacles to the envioronment. 
+        Updates the renderer and the map kept by the laser scaner for each vehicle in the simulator
+
+        Args:
+            n_obstacles (int): number of obstacles to add
+            obstacle_size (list(2)): rectangular size of obstacles
+            
+        Returns:
+            None
+        """
+        if self.renderer is None:
+            self.render()
+        map_img = np.copy(self.empty_map_img)
+
+        obs_size_m = np.array(obstacle_size)
+        obs_size_px = obs_size_m / self.map_resolution
+
+        obs_locations = []
+        while len(obs_locations) < n_obstacles:
+            rand_x = int(np.random.random() * (self.map_width - obs_size_px[0]))
```

Comment:
I think a method to randomize the location of obstacles based on the centerline of the actual map might be more effective. Randomizing it that way you can guarantee the random locations are actually inside the map on the track.

Reply:
I did consider this. 
One of my thoughts was that using the centerline requires that there is a file with the centerline of the track and the widths which can be used. Admittedly, most people do have these files for the standard maps, but it does mean that people who don't have the centerline of a race track can't add obstacles.

Do you think that I should add a separate method that adds obstacles according to the centerline? Or should I replace the code in this method to work from the centerline? Or should I leave it the way that it is?

I think that I will add another method and then people can choose how they want to add obstacles.